aws/resource_aws_instance_test.go

```diff
@@ -3971,6 +4016,33 @@ resource "aws_instance" "foo" {
 `, rInt)
 }
 
+func testAccInstanceConfig_creditSpecification_unknownCpuCredits(rInt int, instance_type string) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "my_vpc" {
+  cidr_block = "172.16.0.0/16"
+
+  tags = {
+    Name = "tf-acctest-%d"
+  }
+}
+
+resource "aws_subnet" "my_subnet" {
+  vpc_id            = "${aws_vpc.my_vpc.id}"
+  cidr_block        = "172.16.20.0/24"
+  availability_zone = "us-west-2a"
+}
+
+resource "aws_instance" "foo" {
+  ami           = "ami-51537029"               # us-west-2
+  instance_type = %q
+  subnet_id     = "${aws_subnet.my_subnet.id}"
+
+  credit_specification {
+  }
+}
+`, rInt, instance_type)
```

Comment:
```suggestion
`, rInt, instance)
```

Reply:
Having it match the attribute name makes the most sense to me.

Reply:
Drive by comment: we historically have preferred naming these consistent with Terraform configuration argument and using [mixed caps to match the Effective Go documentation](https://golang.org/doc/effective_go.html#mixed-caps). So in this case: `instanceType` 👍 

[`golint`](https://github.com/golang/lint) would catch underscored variable names like this, e.g. one example on master:

```
aws/config_test.go:43:5: don't use underscores in Go names; var `test_ec2_describeAccountAttributes_response` should be `testEc2DescribeAccountAttributesResponse` (golint)
var test_ec2_describeAccountAttributes_response = `<DescribeAccountAttributesResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
    ^
```

However there are a _lot_ of linting failures we generally do not want to concern ourselves within `golint`, including its non-configurable initialisms and the underscore check for acceptance test function names (among others), which is why its not enabled by default in our pull request `golangci-lint` testing.

Reply:
Submitted a pull request to the contributing guide to ensure it includes a link to Effective Go: #9031

Reply:
Drive by commenting - love it. Thanks for the additional context @bflad and good call out on the Effective Go resource.